CharlieBackend.Business/Services/AccountService.cs

```diff
@@ -72,7 +72,14 @@ public async Task<Result<AccountDto>> GetAccountCredentialsAsync(AuthenticationD
             {
                 authenticationModel.Password = PasswordHelper.HashPassword(authenticationModel.Password, salt);
 
-                var foundAccount = _mapper.Map<AccountDto>(await _unitOfWork.AccountRepository.GetAccountCredentials(authenticationModel));
+                var accountCredentials = await _unitOfWork.AccountRepository.GetAccountCredentials(authenticationModel);
+
+                if(accountCredentials == null)
+                {
+                    return Result<AccountDto>.GetError(ErrorCode.Unauthorized, "Email or password is incorrect.");
```

Comment:
Have you deleted the same validation in Account Controller?

Reply:
Validation layer is implemented by Business Services. I don't see the same validation in AccontController.

Reply:
This code you wiil see in Account Controller  in SingIn method .I think it will be better if you move Validation  for IsActive and delete (foundAccount == null).       
 if (foundAccount == null)
            {
                return Unauthorized("Incorrect credentials, please try again.");
            }

            if (!foundAccount.IsActive)
            {
                return StatusCode(401, "Account is not active!");
            }

Reply:
We've decided to leave this for further development, when all validation will be placed in extra layer.